x/stake/keeper/key.go

```diff
@@ -0,0 +1,188 @@
+
+//nolint
+var (
+	// Keys for store prefixes
+	ParamKey                         = []byte{0x00} // key for parameters relating to staking
+	PoolKey                          = []byte{0x01} // key for the staking pools
+	ValidatorsKey                    = []byte{0x02} // prefix for each key to a validator
+	ValidatorsByPubKeyIndexKey       = []byte{0x03} // prefix for each key to a validator index, by pubkey
+	ValidatorsBondedIndexKey         = []byte{0x04} // prefix for each key to a validator index, for bonded validators
+	ValidatorsByPowerIndexKey        = []byte{0x05} // prefix for each key to a validator index, sorted by power
+	ValidatorCliffIndexKey           = []byte{0x06} // key for the validator index of the cliff validator
+	ValidatorPowerCliffKey           = []byte{0x07} // key for the power of the validator on the cliff
+	TendermintUpdatesKey             = []byte{0x08} // prefix for each key to a validator which is being updated
+	IntraTxCounterKey                = []byte{0x09} // key for intra-block tx index
+	DelegationKey                    = []byte{0x0A} // key for a delegation
+	UnbondingDelegationKey           = []byte{0x0B} // key for an unbonding-delegation
+	UnbondingDelegationByValIndexKey = []byte{0x0C} // prefix for each key for an unbonding-delegation, by validator owner
+	RedelegationKey                  = []byte{0x0D} // key for a redelegation
+	RedelegationByValSrcIndexKey     = []byte{0x0E} // prefix for each key for an redelegation, by validator owner
+	RedelegationByValDstIndexKey     = []byte{0x0F} // prefix for each key for an redelegation, by validator owner
+)
+
+const maxDigitsForAccount = 12 // ~220,000,000 atoms created at launch
+
+// get the key for the validator with address
+func GetValidatorKey(ownerAddr sdk.Address) []byte {
+	return append(ValidatorsKey, ownerAddr.Bytes()...)
+}
+
+// get the key for the validator with pubkey
+func GetValidatorByPubKeyIndexKey(pubkey crypto.PubKey) []byte {
+	return append(ValidatorsByPubKeyIndexKey, pubkey.Bytes()...)
+}
+
+// get the key for the current validator group, ordered like tendermint
+func GetValidatorsBondedIndexKey(pk crypto.PubKey) []byte {
+	addr := pk.Address()
+	return append(ValidatorsBondedIndexKey, addr.Bytes()...)
+}
+
+// get the power which is the key for the validator used in the power-store
+func GetValidatorsByPowerIndexKey(validator types.Validator, pool types.Pool) []byte {
+
+	// NOTE the address doesn't need to be stored because counter bytes must always be different
+	return GetValidatorPowerRank(validator, pool)
+}
+
+// get the power of a validator
+func GetValidatorPowerRank(validator types.Validator, pool types.Pool) []byte {
+
+	power := validator.EquivalentBondedShares(pool)
+	powerBytes := []byte(power.ToLeftPadded(maxDigitsForAccount)) // power big-endian (more powerful validators first)
+
+	// TODO ensure that the key will be a readable string.. probably should add seperators and have
```

Comment:
Why does the key need to be a readable string? I would think using shorter keys to save space would matter more?

Reply:
based a conversation from Jae a while back on viewing the store directly. don't totally disagree - but let's maybe address this outside of this PR

Reply:
Cool, https://github.com/cosmos/cosmos-sdk/issues/1365.